packages/babel-plugin/src/utils/ast.tsx

```diff
@@ -1,7 +1,38 @@
 import * as t from '@babel/types';
-import traverse from '@babel/traverse';
+import traverse, { NodePath } from '@babel/traverse';
 import { Metadata } from '../types';
 
+/**
+ * Returns the nodes path including the scope of a parent.
+ * @param node
+ * @param parentPath
+ */
+export const getPathOfNode = <TNode extends {}>(
+  node: TNode,
+  parentPath: NodePath
+): NodePath<TNode> => {
+  let foundPath: NodePath | null = null;
+
+  traverse(
```

Comment:
can we think of any better way to do this?

Reply:
Its difficult I think 😞. I mean we are playing with nodes at each level and not having reference to its path info. We are only keeping reference of top level path in `meta.parentPath`. The way you did @madou is not that bad. We have access to `parentPath`. We can traverse through it and find the node. Do we need to wrap node in `t.expression?` Can't we traverse through `parentPath.node` and check the reference of **node** in `enter` like `path.node === node` and return that **path** 🤔. I think reference will be same 🤔.

Reply:
yeah that's the worst part i find of this TBH

`traverse` will only traverse children - not the immediate passed in node - so i used this as a trick to get around it.

we could do `path.parentPath` instead maybe? 🤔

Reply:
😞 we will get best out of babel as we go on. No worries. We just need to make sure to have as many as tests possible right now.